gtdynamics/optimizer/EqualityConstraint.h

```diff
@@ -134,6 +135,39 @@ class VectorExpressionEquality : public EqualityConstraint {
   size_t dim() const override;
 };
 
+/** Equality constraint that force factor error to be 0. */
+class FactorEquality : public EqualityConstraint {
+ protected:
+  gtsam::NoiseModelFactor::shared_ptr factor_;
+  gtsam::Vector tolerance_;
+
+ public:
+  /**
+   * @brief Constructor.
+   *
+   * @param factor  NoiseModel factor.
+   * @param tolerance   vector representing tolerance in each dimension.
+   */
+  FactorEquality(const gtsam::NoiseModelFactor::shared_ptr& factor,
+                 const gtsam::Vector& tolerance)
+      : factor_(factor), tolerance_(tolerance) {}
+
+  FactorEquality(const gtsam::NoiseModelFactor::shared_ptr& factor)
```

Comment:
Missing docstrings here and all methods below this.

Reply:
Done, functions below are all inherited.